plugin/ochttp/server.go

```diff
@@ -223,8 +191,231 @@ func (t *trackingResponseWriter) WriteHeader(statusCode int) {
 	t.statusLine = http.StatusText(t.statusCode)
 }
 
-func (t *trackingResponseWriter) Flush() {
-	if flusher, ok := t.writer.(http.Flusher); ok {
-		flusher.Flush()
+// wrappedResponseWriter returns a wrapped version of the original
+//  ResponseWriter and only implements the same combination of additional
+// interfaces as the original.
+// This implementation is based on https://github.com/felixge/httpsnoop.
+func (t *trackingResponseWriter) wrappedResponseWriter() http.ResponseWriter {
```

Comment:
Is this code generated? It would be great to get the script used committed as well. I can imagine this being useful for databases/sql stuff as well.

Reply:
It wasn't generated, but it could be. I might build a generic generator for optional interfaces as I have some ideas about it. database/sql doesn't have many optional interfaces that need to be wrapped like this. Most of the optional interfaces allow the method implementations to return `driver.ErrSkip` to signal the fast path / implementation is unavailable so I only had to use this logic for `driver.StmtExecContext` and `driver.StmtQueryContext`

Reply:
ack